### PR TITLE
move context out of options bag

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by vhvb1989 — COMMENTED** [review]
I'm surprised there's no tests for using options with context to be updated! 

It would be good to have tests for that to ensure that the client is passing the context to the Internals and it can be cancelled.

Not having any test can lead us to a broken feature 💔

File: sdk/storage/azure-storage-blobs/src/blob_client.cpp [hunk 1]
```diff
@@ -666,6 +666,7 @@ namespace Azure { namespace Storage { namespace Blobs {
       const UndeleteBlobOptions& options,
       const Azure::Core::Context& context) const
   {
+    unused(options);
```
[the remark this answers is not in the document]
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/src/blob_client.cpp:669]: We have an issue tracking that https://github.com/Azure/azure-sdk-for-cpp/issues/1483.

I think i'm gonna replace all `unused` with `void()` in one shot sometime. Before that I just want to keep things consistent within storage sdk.

**Review by katmsft — APPROVED** [review]

**Pull request merged**